pytorch_lightning/trainer/connectors/profiler_connector.py

```diff
@@ -56,4 +57,16 @@ def on_trainer_init(self, profiler: Union[BaseProfiler, str]):
 
     def on_train_start(self, trainer):
         local_rank = trainer.local_rank if trainer.world_size > 1 else None
-        self.trainer.profiler.on_train_start(local_rank)
+        log_dir = None
+        if isinstance(trainer.logger, LoggerCollection):
+            for logger in trainer.logger._logger_iterable:
+                if isinstance(logger, TensorBoardLogger):
+                    log_dir = logger.log_dir
+                    break
```

Comment:
shouldn't we use `trainer.log_dir` directly? It's a property that resolves the log dir.
I think it would be cleaner.
we can update it to also work with the collection.